src/cloud-search-index-storage.js

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-disable lines-between-class-members */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-console */
+import algoliasearch from 'algoliasearch';
+import crypto from 'crypto';
+import clone from 'clone';
+
+export class CloudSearchIndexStorage {
+  #FIELDS_TO_INDEX = new Set(['objectID', 'assetIdentity', 'contentHash', 'sourceName', 'sourceType', 'thumbnailHash', 'file', 'companyId', 'spaceId', 'type', 'tags', 'ocrTags', 'caption', 'width', 'height', 'color', 'sourceId', 'sourceUrl', 'sourceAssetId', 'sourcePath']);
+  #INDEX_PREFIX = 'company';
+
+  constructor(context, companyId) {
+    // TODO: Job document should specify the companyId which should be used as the index name
+    this.indexName = context.env.ALGOLIA_CI_INDEX || `${this.#INDEX_PREFIX}-${companyId}`;
+    console.log('Using Search Index', this.indexName);
+    this.index = algoliasearch(
+      context.env.ALGOLIA_APP_NAME,
+      context.env.ALGOLIA_API_KEY,
+    ).initIndex(this.indexName);
+  }
+
+  withLog(logger) {
+    this.log = logger;
+    return this;
+  }
+
+  /**
+   * Strong Existence Check
+   *
+   * Needs to have both the sourceId and the contentHash (source binary hash) matching
+   * @param {Object} query
+   * @returns Object of hits if there is a hit or undefined if it doesn't exist
```

Comment:
I don't love the ambiguity of the return. It looks like the return can be undefined, null or the hits. 

I'd expect with a name like exists it would only return a boolean.

Reply:
Yeah i see what you mean. The hits are needed for the ingestor: https://github.com/adobe/content-lake-ingestor-service/blob/main/src/ingestor.js#L184-L191

Maybe we call this method something different, something more generic like `getHits`. And the strongExistenceCheck in the ingestor can throw an error if `contentHash` is not defined.